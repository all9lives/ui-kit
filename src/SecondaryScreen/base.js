import { css } from 'styled-components'

const base = css`
  background-color: transparent;
`

export default base
