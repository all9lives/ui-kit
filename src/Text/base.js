import { css } from 'styled-components'
import Unit from '../utils/Unit'
import Size from '../utils/Size'
import Color from '../utils/Color'

const base = css`
  font-size: ${props => Unit.convert(Size[props.size] || props.size)};
  color: ${props => (Color[props.color] || props.color)};
  margin: ${props => Unit.parse(props.margin, (margin) => margin * Size.spacing)};
  align-self: ${props => props.alignSelf};
`

export default base
