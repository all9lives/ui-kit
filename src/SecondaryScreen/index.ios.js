import SecondaryScreen, { Header } from './index.android'

export { Header }
export default SecondaryScreen
