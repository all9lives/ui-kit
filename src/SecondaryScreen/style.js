import StyleSheet from '../utils/StyleSheet'
import Size from '../utils/Size'
import Color from '../utils/Color'

export default new StyleSheet({
  container: {
    height: '100%',
    width: '100%'
  },
  header: {
    backgroundColor: Color.white,
    ios: {
      paddingTop: Size.statusBarSize
    }
  },
  headerInverse: {
    backgroundColor: Color.black
  },
  title: {
    color: Color.black
  },
  titleInverse: {
    color: Color.white
  },
  scrollView: {
    flex: 1
  },
  backBtn: {
    paddingTop: Size.spacing,
    paddingBottom: Size.spacing,
    paddingLeft: Size.spacing,
    paddingRight: Size.spacing
  }
})
