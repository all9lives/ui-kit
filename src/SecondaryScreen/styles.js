import { Platform } from 'react-native'
import StyleSheet from '../utils/StyleSheet'
import Shadow from '../utils/Shadow'
import Color from '../utils/Color'
import Size from '../utils/Size'

export const STATUS_BAR_HEIGHT = Platform.OS === 'ios' ? 20 : 0
export const HEADER_HEIGHT = 45
export const TITLE_HEIGHT = 40
export const TITLE_BEGIN_Y = HEADER_HEIGHT + STATUS_BAR_HEIGHT + Size.spacing
export const TITLE_END_Y = STATUS_BAR_HEIGHT
export const SCREEN_PADDING = Size.spacing * 2
export const TITLE_BEGIN_X = SCREEN_PADDING
export const TITLE_END_X = HEADER_HEIGHT

export default new StyleSheet({
  container: {
    height: '100%',
    width: '100%',
    ios: {
      paddingTop: STATUS_BAR_HEIGHT
    }
  },
  header: {
    position: 'absolute',
    top: 0,
    left: 0,
    right: 0,
    height: HEADER_HEIGHT,
    backgroundColor: Color.white,
    zIndex: 1,
    ios: {
      top: STATUS_BAR_HEIGHT
    }
  },
  shadow: {
    ...Shadow.normal()
  },
  scrollable: {
    flex: 1
  },
  body: {
    paddingTop: HEADER_HEIGHT + TITLE_HEIGHT + 5 * Size.spacing
  },
  backBtn: {
    width: HEADER_HEIGHT,
    height: HEADER_HEIGHT
  },
  actionBtns: {
    paddingRight: SCREEN_PADDING / 2
  },
  actionBtn: {
    marginLeft: Size.spacing,
    marginRight: Size.spacing
  },
  title: {
    height: HEADER_HEIGHT,
    justifyContent: 'center',
    position: 'absolute',
    zIndex: 9,
    android: {
      elevation: 6
    }
  },
  titleText: {
    backgroundColor: 'transparent'
  }
})
