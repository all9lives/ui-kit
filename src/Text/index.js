import React from 'react';
import { Text as PrimitiveText } from 'react-primitives';

const Text = ({ children }) => (
  <PrimitiveText>{children}</PrimitiveText>
);

export default Text;
