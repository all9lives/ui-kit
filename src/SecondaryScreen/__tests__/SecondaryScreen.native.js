/* eslint-env jest */
import React from 'react'
import { shallow } from 'enzyme'
import SecondaryScreen, { Base, Header } from '../index.ios.js'
import Text from '../../Text'

jest.mock('Animated', () => {
  const ActualAnimated = require.requireActual('Animated')
  return {
    ...ActualAnimated,
    timing: (value, config) => {
      return {
        start: (callback) => {
          value.setValue(config.toValue)
          callback && callback()
        }
      }
    }
  }
})

const diveToBase = wrapper => wrapper.find(Base).dive()

describe('SecondaryScreen.native', () => {
  it('renders a snapshot', () => {
    const wrapper = shallow(
      <SecondaryScreen><Text>SecondaryScreen</Text></SecondaryScreen>
    )
    expect(diveToBase(wrapper)).toMatchSnapshot()
  })

  it('renders a snapshot with `title:MyTitle`', () => {
    const wrapper = shallow(
      <SecondaryScreen title='MyTitle'>
        <Text>SecondaryScreen</Text>
      </SecondaryScreen>
    )
    expect(diveToBase(wrapper)).toMatchSnapshot()
  })

  it('renders a snapshot with `actionButtons`', () => {
    const wrapper = shallow(
      <SecondaryScreen actionButtons={[<Text>1</Text>, <Text>2</Text>]}>
        <Text>SecondaryScreen</Text>
      </SecondaryScreen>
    )
    expect(diveToBase(wrapper)).toMatchSnapshot()
  })

  it('renders a snapshot with `onBackPress`', () => {
    const wrapper = shallow(
      <SecondaryScreen onBackPress={() => console.log('onBackPressed')}>
        <Text>SecondaryScreen</Text>
      </SecondaryScreen>
    )
    expect(diveToBase(wrapper)).toMatchSnapshot()
  })

  it('renders a snapshot with `whiteStyle`', () => {
    const wrapper = shallow(
      <SecondaryScreen whiteStyle>
        <Text>SecondaryScreen</Text>
      </SecondaryScreen>
    )
    expect(diveToBase(wrapper)).toMatchSnapshot()
  })

  it('renders a snapshot with `style`', () => {
    const wrapper = shallow(
      <SecondaryScreen style={{ backgroundColor: 'black' }}>
        <Text>SecondaryScreen</Text>
      </SecondaryScreen>
    )
    expect(diveToBase(wrapper)).toMatchSnapshot()
  })

  it('renders a snapshot with `headerStyle`', () => {
    const wrapper = shallow(
      <SecondaryScreen headerStyle={{ backgroundColor: 'black' }}>
        <Text>SecondaryScreen</Text>
      </SecondaryScreen>
    )
    expect(diveToBase(wrapper)).toMatchSnapshot()
  })

  it('renders a snapshot with `bodyStyle`', () => {
    const wrapper = shallow(
      <SecondaryScreen bodyStyle={{ backgroundColor: 'black' }}>
        <Text>SecondaryScreen</Text>
      </SecondaryScreen>
    )
    expect(diveToBase(wrapper)).toMatchSnapshot()
  })

  it('renders a snapshot with `onRefresh`', () => {
    const wrapper = shallow(
      <SecondaryScreen onRefresh={() => {}}>
        <Text>SecondaryScreen</Text>
      </SecondaryScreen>
    )
    expect(diveToBase(wrapper)).toMatchSnapshot()
  })

  it('renders a snapshot with `refreshing`', () => {
    const wrapper = shallow(
      <SecondaryScreen refreshing>
        <Text>SecondaryScreen</Text>
      </SecondaryScreen>
    )
    expect(diveToBase(wrapper)).toMatchSnapshot()
  })

  it('renders a snapshot with `handleScroll:toggleHeader:true`', () => {
    const wrapper = shallow(
      <SecondaryScreen><Text>SecondaryScreen</Text></SecondaryScreen>
    )
    const baseWrapper = diveToBase(wrapper)
    baseWrapper.instance().handleScroll({ nativeEvent: { contentOffset: { y: 40 } } })
    baseWrapper.update()
    expect(baseWrapper).toMatchSnapshot()
  })

  it('renders a snapshot with `handleScroll:toggleHeader:false`', () => {
    const wrapper = shallow(
      <SecondaryScreen><Text>SecondaryScreen</Text></SecondaryScreen>
    )
    const baseWrapper = diveToBase(wrapper)
    baseWrapper.instance().setState({ toggleHeader: true })
    baseWrapper.update()
    baseWrapper.instance().handleScroll({ nativeEvent: { contentOffset: { y: 30 } } })
    expect(baseWrapper).toMatchSnapshot()
  })

  it('renders a snapshot with `handleScroll:nothing`', () => {
    const wrapper = shallow(
      <SecondaryScreen><Text>SecondaryScreen</Text></SecondaryScreen>
    )
    const baseWrapper = diveToBase(wrapper)
    baseWrapper.instance().handleScroll({ nativeEvent: { contentOffset: { y: 30 } } })
    expect(baseWrapper).toMatchSnapshot()
  })

  it('renders a snapshot with `componentWillUpdate`', () => {
    const wrapper = shallow(
      <SecondaryScreen><Text>SecondaryScreen</Text></SecondaryScreen>
    )
    const baseWrapper = diveToBase(wrapper)
    baseWrapper.setState({ toggleHeader: false })
    baseWrapper.update()
    expect(baseWrapper).toMatchSnapshot()
  })
})

describe('SecondaryScreen.Header.native', () => {
  it('renders a snapshot', () => {
    const wrapper = shallow(
      <Header actionButtons={[]} />
    )
    expect(wrapper).toMatchSnapshot()
  })

  it('renders a snapshot with `toggleHeader:true`', () => {
    const wrapper = shallow(
      <Header toggleHeader actionButtons={[<Text>Action</Text>]} />
    )
    expect(wrapper).toMatchSnapshot()
  })

  it('renders a snapshot with `whiteStyle:true`', () => {
    const wrapper = shallow(
      <Header actionButtons={[]} whiteStyle />
    )
    expect(wrapper).toMatchSnapshot()
  })
})
