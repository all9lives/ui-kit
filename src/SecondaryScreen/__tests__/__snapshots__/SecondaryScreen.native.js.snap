// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SecondaryScreen.Header.native renders a snapshot 1`] = `
<Styled(View)
  align="middle"
  direction="row"
  margin={0}
  padding={0}
  style={
    Array [
      Object {
        "backgroundColor": "#fafafa",
        "height": 45,
        "left": 0,
        "position": "absolute",
        "right": 0,
        "top": 20,
        "zIndex": 1,
      },
      undefined,
      undefined,
    ]
  }
>
  <TouchableOpacity
    activeOpacity={0.2}
  >
    <Styled(View)
      align="center middle"
      direction="column"
      margin={0}
      padding={0}
      style={
        Object {
          "height": 45,
          "width": 45,
        }
      }
    >
      <Base
        size="regular"
        source={1}
      />
    </Styled(View)>
  </TouchableOpacity>
  <Styled(View)
    align=""
    direction="column"
    grow={1}
    margin={0}
    padding={0}
  />
  <Styled(View)
    align="middle"
    direction="row"
    margin={0}
    padding={0}
    style={
      Object {
        "paddingRight": 10,
      }
    }
  />
</Styled(View)>
`;

exports[`SecondaryScreen.Header.native renders a snapshot with \`toggleHeader:true\` 1`] = `
<Styled(View)
  align="middle"
  direction="row"
  margin={0}
  padding={0}
  style={
    Array [
      Object {
        "backgroundColor": "#fafafa",
        "height": 45,
        "left": 0,
        "position": "absolute",
        "right": 0,
        "top": 20,
        "zIndex": 1,
      },
      Object {
        "shadowColor": "black",
        "shadowOffset": Object {
          "height": 5,
          "width": 5,
        },
        "shadowOpacity": 0.2,
        "shadowRadius": 2.5,
      },
      undefined,
    ]
  }
>
  <TouchableOpacity
    activeOpacity={0.2}
  >
    <Styled(View)
      align="center middle"
      direction="column"
      margin={0}
      padding={0}
      style={
        Object {
          "height": 45,
          "width": 45,
        }
      }
    >
      <Base
        size="regular"
        source={1}
      />
    </Styled(View)>
  </TouchableOpacity>
  <Styled(View)
    align=""
    direction="column"
    grow={1}
    margin={0}
    padding={0}
  />
  <Styled(View)
    align="middle"
    direction="row"
    margin={0}
    padding={0}
    style={
      Object {
        "paddingRight": 10,
      }
    }
  >
    <Styled(View)
      align=""
      direction="column"
      key="0"
      margin={0}
      padding={0}
      style={
        Object {
          "marginLeft": 10,
          "marginRight": 10,
        }
      }
    >
      <Styled(Text)
        alignSelf="auto"
        color="#191919"
        margin={0}
        size={16}
      >
        Action
      </Styled(Text)>
    </Styled(View)>
  </Styled(View)>
</Styled(View)>
`;

exports[`SecondaryScreen.Header.native renders a snapshot with \`whiteBackIcon:true\` 1`] = `
<Styled(View)
  align="middle"
  direction="row"
  margin={0}
  padding={0}
  style={
    Array [
      Object {
        "backgroundColor": "#fafafa",
        "height": 45,
        "left": 0,
        "position": "absolute",
        "right": 0,
        "top": 20,
        "zIndex": 1,
      },
      undefined,
      undefined,
    ]
  }
>
  <TouchableOpacity
    activeOpacity={0.2}
  >
    <Styled(View)
      align="center middle"
      direction="column"
      margin={0}
      padding={0}
      style={
        Object {
          "height": 45,
          "width": 45,
        }
      }
    >
      <Base
        size="regular"
        source={1}
      />
    </Styled(View)>
  </TouchableOpacity>
  <Styled(View)
    align=""
    direction="column"
    grow={1}
    margin={0}
    padding={0}
  />
  <Styled(View)
    align="middle"
    direction="row"
    margin={0}
    padding={0}
    style={
      Object {
        "paddingRight": 10,
      }
    }
  />
</Styled(View)>
`;

exports[`SecondaryScreen.native renders a snapshot 1`] = `
<Styled(View)
  align=""
  direction="column"
  margin={0}
  padding={0}
  style={
    Array [
      Array [
        Object {
          "backgroundColor": "transparent",
        },
        undefined,
      ],
      Object {
        "height": "100%",
        "paddingTop": 20,
        "width": "100%",
      },
    ]
  }
>
  <Header
    actionButtons={Array []}
    toggleHeader={false}
  />
  <AnimatedComponent
    style={
      Array [
        Object {
          "height": 45,
          "justifyContent": "center",
          "position": "absolute",
          "zIndex": 9,
        },
        Object {
          "left": 20,
          "top": 75,
        },
      ]
    }
  >
    <AnimatedComponent
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
          },
          Object {
            "fontSize": 25,
          },
        ]
      }
    >
      Title
    </AnimatedComponent>
  </AnimatedComponent>
  <ScrollViewMock
    contentContainerStyle={
      Array [
        Object {
          "paddingTop": 135,
        },
        undefined,
      ]
    }
    onScroll={[Function]}
    scrollEventThrottle={16}
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <Styled(Text)
      alignSelf="auto"
      color="#191919"
      margin={0}
      size={16}
    >
      SecondaryScreen
    </Styled(Text)>
  </ScrollViewMock>
</Styled(View)>
`;

exports[`SecondaryScreen.native renders a snapshot with \`actionButtons\` 1`] = `
<Styled(View)
  align=""
  direction="column"
  margin={0}
  padding={0}
  style={
    Array [
      Array [
        Object {
          "backgroundColor": "transparent",
        },
        undefined,
      ],
      Object {
        "height": "100%",
        "paddingTop": 20,
        "width": "100%",
      },
    ]
  }
>
  <Header
    actionButtons={
      Array [
        <Styled(Text)
          alignSelf="auto"
          color="#191919"
          margin={0}
          size={16}
        >
          1
        </Styled(Text)>,
        <Styled(Text)
          alignSelf="auto"
          color="#191919"
          margin={0}
          size={16}
        >
          2
        </Styled(Text)>,
      ]
    }
    toggleHeader={false}
  />
  <AnimatedComponent
    style={
      Array [
        Object {
          "height": 45,
          "justifyContent": "center",
          "position": "absolute",
          "zIndex": 9,
        },
        Object {
          "left": 20,
          "top": 75,
        },
      ]
    }
  >
    <AnimatedComponent
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
          },
          Object {
            "fontSize": 25,
          },
        ]
      }
    >
      Title
    </AnimatedComponent>
  </AnimatedComponent>
  <ScrollViewMock
    contentContainerStyle={
      Array [
        Object {
          "paddingTop": 135,
        },
        undefined,
      ]
    }
    onScroll={[Function]}
    scrollEventThrottle={16}
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <Styled(Text)
      alignSelf="auto"
      color="#191919"
      margin={0}
      size={16}
    >
      SecondaryScreen
    </Styled(Text)>
  </ScrollViewMock>
</Styled(View)>
`;

exports[`SecondaryScreen.native renders a snapshot with \`bodyStyle\` 1`] = `
<Styled(View)
  align=""
  direction="column"
  margin={0}
  padding={0}
  style={
    Array [
      Array [
        Object {
          "backgroundColor": "transparent",
        },
        undefined,
      ],
      Object {
        "height": "100%",
        "paddingTop": 20,
        "width": "100%",
      },
    ]
  }
>
  <Header
    actionButtons={Array []}
    toggleHeader={false}
  />
  <AnimatedComponent
    style={
      Array [
        Object {
          "height": 45,
          "justifyContent": "center",
          "position": "absolute",
          "zIndex": 9,
        },
        Object {
          "left": 20,
          "top": 75,
        },
      ]
    }
  >
    <AnimatedComponent
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
          },
          Object {
            "fontSize": 25,
          },
        ]
      }
    >
      Title
    </AnimatedComponent>
  </AnimatedComponent>
  <ScrollViewMock
    contentContainerStyle={
      Array [
        Object {
          "paddingTop": 135,
        },
        Object {
          "backgroundColor": "black",
        },
      ]
    }
    onScroll={[Function]}
    scrollEventThrottle={16}
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <Styled(Text)
      alignSelf="auto"
      color="#191919"
      margin={0}
      size={16}
    >
      SecondaryScreen
    </Styled(Text)>
  </ScrollViewMock>
</Styled(View)>
`;

exports[`SecondaryScreen.native renders a snapshot with \`componentWillUpdate\` 1`] = `
<Styled(View)
  align=""
  direction="column"
  margin={0}
  padding={0}
  style={
    Array [
      Array [
        Object {
          "backgroundColor": "transparent",
        },
        undefined,
      ],
      Object {
        "height": "100%",
        "paddingTop": 20,
        "width": "100%",
      },
    ]
  }
>
  <Header
    actionButtons={Array []}
    toggleHeader={false}
  />
  <AnimatedComponent
    style={
      Array [
        Object {
          "height": 45,
          "justifyContent": "center",
          "position": "absolute",
          "zIndex": 9,
        },
        Object {
          "left": 20,
          "top": 75,
        },
      ]
    }
  >
    <AnimatedComponent
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
          },
          Object {
            "fontSize": 25,
          },
        ]
      }
    >
      Title
    </AnimatedComponent>
  </AnimatedComponent>
  <ScrollViewMock
    contentContainerStyle={
      Array [
        Object {
          "paddingTop": 135,
        },
        undefined,
      ]
    }
    onScroll={[Function]}
    scrollEventThrottle={16}
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <Styled(Text)
      alignSelf="auto"
      color="#191919"
      margin={0}
      size={16}
    >
      SecondaryScreen
    </Styled(Text)>
  </ScrollViewMock>
</Styled(View)>
`;

exports[`SecondaryScreen.native renders a snapshot with \`handleScroll:nothing\` 1`] = `
<Styled(View)
  align=""
  direction="column"
  margin={0}
  padding={0}
  style={
    Array [
      Array [
        Object {
          "backgroundColor": "transparent",
        },
        undefined,
      ],
      Object {
        "height": "100%",
        "paddingTop": 20,
        "width": "100%",
      },
    ]
  }
>
  <Header
    actionButtons={Array []}
    toggleHeader={false}
  />
  <AnimatedComponent
    style={
      Array [
        Object {
          "height": 45,
          "justifyContent": "center",
          "position": "absolute",
          "zIndex": 9,
        },
        Object {
          "left": 20,
          "top": 75,
        },
      ]
    }
  >
    <AnimatedComponent
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
          },
          Object {
            "fontSize": 25,
          },
        ]
      }
    >
      Title
    </AnimatedComponent>
  </AnimatedComponent>
  <ScrollViewMock
    contentContainerStyle={
      Array [
        Object {
          "paddingTop": 135,
        },
        undefined,
      ]
    }
    onScroll={[Function]}
    scrollEventThrottle={16}
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <Styled(Text)
      alignSelf="auto"
      color="#191919"
      margin={0}
      size={16}
    >
      SecondaryScreen
    </Styled(Text)>
  </ScrollViewMock>
</Styled(View)>
`;

exports[`SecondaryScreen.native renders a snapshot with \`handleScroll:toggleHeader:false\` 1`] = `
<Styled(View)
  align=""
  direction="column"
  margin={0}
  padding={0}
  style={
    Array [
      Array [
        Object {
          "backgroundColor": "transparent",
        },
        undefined,
      ],
      Object {
        "height": "100%",
        "paddingTop": 20,
        "width": "100%",
      },
    ]
  }
>
  <Header
    actionButtons={Array []}
    toggleHeader={true}
  />
  <AnimatedComponent
    style={
      Array [
        Object {
          "height": 45,
          "justifyContent": "center",
          "position": "absolute",
          "zIndex": 9,
        },
        Object {
          "left": 20,
          "top": 75,
        },
      ]
    }
  >
    <AnimatedComponent
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
          },
          Object {
            "fontSize": 25,
          },
        ]
      }
    >
      Title
    </AnimatedComponent>
  </AnimatedComponent>
  <ScrollViewMock
    contentContainerStyle={
      Array [
        Object {
          "paddingTop": 135,
        },
        undefined,
      ]
    }
    onScroll={[Function]}
    scrollEventThrottle={16}
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <Styled(Text)
      alignSelf="auto"
      color="#191919"
      margin={0}
      size={16}
    >
      SecondaryScreen
    </Styled(Text)>
  </ScrollViewMock>
</Styled(View)>
`;

exports[`SecondaryScreen.native renders a snapshot with \`handleScroll:toggleHeader:true\` 1`] = `
<Styled(View)
  align=""
  direction="column"
  margin={0}
  padding={0}
  style={
    Array [
      Array [
        Object {
          "backgroundColor": "transparent",
        },
        undefined,
      ],
      Object {
        "height": "100%",
        "paddingTop": 20,
        "width": "100%",
      },
    ]
  }
>
  <Header
    actionButtons={Array []}
    toggleHeader={true}
  />
  <AnimatedComponent
    style={
      Array [
        Object {
          "height": 45,
          "justifyContent": "center",
          "position": "absolute",
          "zIndex": 9,
        },
        Object {
          "left": 45,
          "top": 20,
        },
      ]
    }
  >
    <AnimatedComponent
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
          },
          Object {
            "fontSize": 16,
          },
        ]
      }
    >
      Title
    </AnimatedComponent>
  </AnimatedComponent>
  <ScrollViewMock
    contentContainerStyle={
      Array [
        Object {
          "paddingTop": 135,
        },
        undefined,
      ]
    }
    onScroll={[Function]}
    scrollEventThrottle={16}
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <Styled(Text)
      alignSelf="auto"
      color="#191919"
      margin={0}
      size={16}
    >
      SecondaryScreen
    </Styled(Text)>
  </ScrollViewMock>
</Styled(View)>
`;

exports[`SecondaryScreen.native renders a snapshot with \`headerStyle\` 1`] = `
<Styled(View)
  align=""
  direction="column"
  margin={0}
  padding={0}
  style={
    Array [
      Array [
        Object {
          "backgroundColor": "transparent",
        },
        undefined,
      ],
      Object {
        "height": "100%",
        "paddingTop": 20,
        "width": "100%",
      },
    ]
  }
>
  <Header
    actionButtons={Array []}
    headerStyle={
      Object {
        "backgroundColor": "black",
      }
    }
    toggleHeader={false}
  />
  <AnimatedComponent
    style={
      Array [
        Object {
          "height": 45,
          "justifyContent": "center",
          "position": "absolute",
          "zIndex": 9,
        },
        Object {
          "left": 20,
          "top": 75,
        },
      ]
    }
  >
    <AnimatedComponent
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
          },
          Object {
            "fontSize": 25,
          },
        ]
      }
    >
      Title
    </AnimatedComponent>
  </AnimatedComponent>
  <ScrollViewMock
    contentContainerStyle={
      Array [
        Object {
          "paddingTop": 135,
        },
        undefined,
      ]
    }
    onScroll={[Function]}
    scrollEventThrottle={16}
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <Styled(Text)
      alignSelf="auto"
      color="#191919"
      margin={0}
      size={16}
    >
      SecondaryScreen
    </Styled(Text)>
  </ScrollViewMock>
</Styled(View)>
`;

exports[`SecondaryScreen.native renders a snapshot with \`onBackPress\` 1`] = `
<Styled(View)
  align=""
  direction="column"
  margin={0}
  padding={0}
  style={
    Array [
      Array [
        Object {
          "backgroundColor": "transparent",
        },
        undefined,
      ],
      Object {
        "height": "100%",
        "paddingTop": 20,
        "width": "100%",
      },
    ]
  }
>
  <Header
    actionButtons={Array []}
    onBackPress={[Function]}
    toggleHeader={false}
  />
  <AnimatedComponent
    style={
      Array [
        Object {
          "height": 45,
          "justifyContent": "center",
          "position": "absolute",
          "zIndex": 9,
        },
        Object {
          "left": 20,
          "top": 75,
        },
      ]
    }
  >
    <AnimatedComponent
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
          },
          Object {
            "fontSize": 25,
          },
        ]
      }
    >
      Title
    </AnimatedComponent>
  </AnimatedComponent>
  <ScrollViewMock
    contentContainerStyle={
      Array [
        Object {
          "paddingTop": 135,
        },
        undefined,
      ]
    }
    onScroll={[Function]}
    scrollEventThrottle={16}
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <Styled(Text)
      alignSelf="auto"
      color="#191919"
      margin={0}
      size={16}
    >
      SecondaryScreen
    </Styled(Text)>
  </ScrollViewMock>
</Styled(View)>
`;

exports[`SecondaryScreen.native renders a snapshot with \`onRefresh\` 1`] = `
<Styled(View)
  align=""
  direction="column"
  margin={0}
  padding={0}
  style={
    Array [
      Array [
        Object {
          "backgroundColor": "transparent",
        },
        undefined,
      ],
      Object {
        "height": "100%",
        "paddingTop": 20,
        "width": "100%",
      },
    ]
  }
>
  <Header
    actionButtons={Array []}
    toggleHeader={false}
  />
  <AnimatedComponent
    style={
      Array [
        Object {
          "height": 45,
          "justifyContent": "center",
          "position": "absolute",
          "zIndex": 9,
        },
        Object {
          "left": 20,
          "top": 75,
        },
      ]
    }
  >
    <AnimatedComponent
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
          },
          Object {
            "fontSize": 25,
          },
        ]
      }
    >
      Title
    </AnimatedComponent>
  </AnimatedComponent>
  <ScrollViewMock
    contentContainerStyle={
      Array [
        Object {
          "paddingTop": 135,
        },
        undefined,
      ]
    }
    onScroll={[Function]}
    refreshControl={
      <RefreshControlMock
        onRefresh={[Function]}
        refreshing={false}
      />
    }
    scrollEventThrottle={16}
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <Styled(Text)
      alignSelf="auto"
      color="#191919"
      margin={0}
      size={16}
    >
      SecondaryScreen
    </Styled(Text)>
  </ScrollViewMock>
</Styled(View)>
`;

exports[`SecondaryScreen.native renders a snapshot with \`refreshing\` 1`] = `
<Styled(View)
  align=""
  direction="column"
  margin={0}
  padding={0}
  style={
    Array [
      Array [
        Object {
          "backgroundColor": "transparent",
        },
        undefined,
      ],
      Object {
        "height": "100%",
        "paddingTop": 20,
        "width": "100%",
      },
    ]
  }
>
  <Header
    actionButtons={Array []}
    toggleHeader={false}
  />
  <AnimatedComponent
    style={
      Array [
        Object {
          "height": 45,
          "justifyContent": "center",
          "position": "absolute",
          "zIndex": 9,
        },
        Object {
          "left": 20,
          "top": 75,
        },
      ]
    }
  >
    <AnimatedComponent
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
          },
          Object {
            "fontSize": 25,
          },
        ]
      }
    >
      Title
    </AnimatedComponent>
  </AnimatedComponent>
  <ScrollViewMock
    contentContainerStyle={
      Array [
        Object {
          "paddingTop": 135,
        },
        undefined,
      ]
    }
    onScroll={[Function]}
    refreshControl={
      <RefreshControlMock
        onRefresh={undefined}
        refreshing={true}
      />
    }
    scrollEventThrottle={16}
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <Styled(Text)
      alignSelf="auto"
      color="#191919"
      margin={0}
      size={16}
    >
      SecondaryScreen
    </Styled(Text)>
  </ScrollViewMock>
</Styled(View)>
`;

exports[`SecondaryScreen.native renders a snapshot with \`style\` 1`] = `
<Styled(View)
  align=""
  direction="column"
  margin={0}
  padding={0}
  style={
    Array [
      Array [
        Object {
          "backgroundColor": "transparent",
        },
        Object {
          "backgroundColor": "black",
        },
      ],
      Object {
        "height": "100%",
        "paddingTop": 20,
        "width": "100%",
      },
    ]
  }
>
  <Header
    actionButtons={Array []}
    toggleHeader={false}
  />
  <AnimatedComponent
    style={
      Array [
        Object {
          "height": 45,
          "justifyContent": "center",
          "position": "absolute",
          "zIndex": 9,
        },
        Object {
          "left": 20,
          "top": 75,
        },
      ]
    }
  >
    <AnimatedComponent
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
          },
          Object {
            "fontSize": 25,
          },
        ]
      }
    >
      Title
    </AnimatedComponent>
  </AnimatedComponent>
  <ScrollViewMock
    contentContainerStyle={
      Array [
        Object {
          "paddingTop": 135,
        },
        undefined,
      ]
    }
    onScroll={[Function]}
    scrollEventThrottle={16}
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <Styled(Text)
      alignSelf="auto"
      color="#191919"
      margin={0}
      size={16}
    >
      SecondaryScreen
    </Styled(Text)>
  </ScrollViewMock>
</Styled(View)>
`;

exports[`SecondaryScreen.native renders a snapshot with \`title:MyTitle\` 1`] = `
<Styled(View)
  align=""
  direction="column"
  margin={0}
  padding={0}
  style={
    Array [
      Array [
        Object {
          "backgroundColor": "transparent",
        },
        undefined,
      ],
      Object {
        "height": "100%",
        "paddingTop": 20,
        "width": "100%",
      },
    ]
  }
>
  <Header
    actionButtons={Array []}
    toggleHeader={false}
  />
  <AnimatedComponent
    style={
      Array [
        Object {
          "height": 45,
          "justifyContent": "center",
          "position": "absolute",
          "zIndex": 9,
        },
        Object {
          "left": 20,
          "top": 75,
        },
      ]
    }
  >
    <AnimatedComponent
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
          },
          Object {
            "fontSize": 25,
          },
        ]
      }
    >
      MyTitle
    </AnimatedComponent>
  </AnimatedComponent>
  <ScrollViewMock
    contentContainerStyle={
      Array [
        Object {
          "paddingTop": 135,
        },
        undefined,
      ]
    }
    onScroll={[Function]}
    scrollEventThrottle={16}
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <Styled(Text)
      alignSelf="auto"
      color="#191919"
      margin={0}
      size={16}
    >
      SecondaryScreen
    </Styled(Text)>
  </ScrollViewMock>
</Styled(View)>
`;

exports[`SecondaryScreen.native renders a snapshot with \`whiteBackIcon\` 1`] = `
<Styled(View)
  align=""
  direction="column"
  margin={0}
  padding={0}
  style={
    Array [
      Array [
        Object {
          "backgroundColor": "transparent",
        },
        undefined,
      ],
      Object {
        "height": "100%",
        "paddingTop": 20,
        "width": "100%",
      },
    ]
  }
>
  <Header
    actionButtons={Array []}
    toggleHeader={false}
    whiteBackIcon={true}
  />
  <AnimatedComponent
    style={
      Array [
        Object {
          "height": 45,
          "justifyContent": "center",
          "position": "absolute",
          "zIndex": 9,
        },
        Object {
          "left": 20,
          "top": 75,
        },
      ]
    }
  >
    <AnimatedComponent
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
          },
          Object {
            "fontSize": 25,
          },
        ]
      }
    >
      Title
    </AnimatedComponent>
  </AnimatedComponent>
  <ScrollViewMock
    contentContainerStyle={
      Array [
        Object {
          "paddingTop": 135,
        },
        undefined,
      ]
    }
    onScroll={[Function]}
    scrollEventThrottle={16}
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <Styled(Text)
      alignSelf="auto"
      color="#191919"
      margin={0}
      size={16}
    >
      SecondaryScreen
    </Styled(Text)>
  </ScrollViewMock>
</Styled(View)>
`;
